### ISSUE TITLE
Azurerm_container_group deploys a container but sticks on "Still creating..." and eventually errors

### ISSUE THREAD

**Issue by danielearwicker** [hashicorp/terraform-provider-azurerm#17706]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.14.0

### Affected Resource(s)/Data Source(s)

azurerm_container_group 

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
  subscription_id = "<azure-subscription-here>"
}

resource "azurerm_resource_group" "repro" {
  name      = "repro"
  location  = "westeurope"
}

resource "azurerm_user_assigned_identity" "id1" {
  resource_group_name = azurerm_resource_group.repro.name
  location            = azurerm_resource_group.repro.location
  name = "id1"
}

resource "azurerm_user_assigned_identity" "id2" {
  resource_group_name = azurerm_resource_group.repro.name
  location            = azurerm_resource_group.repro.location
  name = "id2"
}

resource "azurerm_user_assigned_identity" "id3" {
  resource_group_name = azurerm_resource_group.repro.name
  location            = azurerm_resource_group.repro.location
  name = "id3"
}

resource "azurerm_container_group" "helloworld" {
  name                = "helloworld"
  location            = azurerm_resource_group.repro.location
  resource_group_name = azurerm_resource_group.repro.name
  ip_address_type     = "Public"
  os_type             = "Linux"

  identity {
    type         = "UserAssigned"
    identity_ids = [
      azurerm_user_assigned_identity.id1.id,
      azurerm_user_assigned_identity.id2.id,
      azurerm_user_assigned_identity.id3.id
    ]
  }

  container {
    name   = "helloworld"
    image  = "nginxdemos/hello"
    cpu    = "4"
    memory = "2"

    ports {
      port     = 80
      protocol = "TCP"
    }

    ports {
      port     = 443
      protocol = "TCP"
    }
  }
}

resource "azurerm_container_group" "goodbyeworld" {
  name                = "goodbyeworld"
  location            = azurerm_resource_group.repro.location
  resource_group_name = azurerm_resource_group.repro.name
  ip_address_type     = "Public"
  os_type             = "Linux"

  identity {
    type         = "UserAssigned"
    identity_ids = [
      azurerm_user_assigned_identity.id1.id,
      azurerm_user_assigned_identity.id2.id,
      azurerm_user_assigned_identity.id3.id
    ]
  }

  container {
    name   = "goodbyeworld"
    image  = "nginxdemos/hello"
    cpu    = "4"
    memory = "2"

    ports {
      port     = 80
      protocol = "TCP"
    }

    ports {
      port     = 443
      protocol = "TCP"
    }
  }
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

Deploys container(s), stops saying "Still creating..." soon after they are up, and `terraform apply` succeeds.

### Actual Behaviour

Output from `terraform apply` ends with:

```
azurerm_container_group.goodbyeworld: Still creating... [28m50s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m0s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m10s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m20s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m30s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m40s elapsed]
azurerm_container_group.goodbyeworld: Still creating... [29m50s elapsed]

Error: creating Container Group (Subscription: "ecbef10d-375a-426a-8a30-8f5b1f6cd139"
Resource Group Name: "repro"
Container Group Name: "goodbyeworld"): polling after ContainerGroupsCreateOrUpdate: Future#WaitForCompletion: context has been cancelled: StatusCode=200 -- Original Error: context deadline exceeded

  on processor.tf line 63, in resource "azurerm_container_group" "goodbyeworld":
  63: resource "azurerm_container_group" "goodbyeworld" {
```

It gave this error at 15:00 UTC, so I started it at 14:30 UTC.

The container is actually up and running fine, the logs show it was up and ready after 2-3 minutes:

```
2022/07/21 14:32:55 [notice] 24#24: start worker processes
2022/07/21 14:32:55 [notice] 24#24: start worker process 53
2022/07/21 14:32:55 [notice] 24#24: start worker process 54
2022/07/21 14:32:55 [notice] 24#24: start worker process 55
2022/07/21 14:32:55 [notice] 24#24: start worker process 56
```

### Steps to Reproduce

The included script deploys 3 azure identities and 2 containers that use them.

I found that if I applied it first with the `goodbyeworld` container group commented out, it took a while to finish the deployment but it worked. And then I uncommented the `goodbyeworld` and re-applied, at which point I reproduced the long delay problem.

I have included the identities to more closely mimic our real deployments, which have been failing like this every time for the last 24 hours, having previously been fine for about 6 months.

Just deploying a container without assigned identities seemed okay.

### Important Factoids

N/A

**Comment by magodo** [comment]
@danielearwicker Thank you for submitting this!

I've tried your above steps, comment the `goodbyeworld` and apply the remainings in the first apply, which takes 2min. Then uncomment the `goodbyeworld` and apply again, which takes 1min. Both applies succeeded.

Would you please reproduce this issue and provide the debug log, so that we can further investigate why it takes so long?

**Comment by danielearwicker** [comment]
Thanks very much for looking. Frustratingly, this morning it is behaving itself for me too, creating multiple extra processors in around 30 seconds.

One other thing I've only just noticed is that when it was failing, although the container itself would be apparently healthy (executing and chatting expected things logs the whole time), the Azure portal shows the Status as `Failed`. But all our attempts this morning have gone to the state `Running`.

So I suspect this is most likely not a TF or azure connector issue but an actual problem in the Azure ACI service itself, which may have now been resolved. 🤞
